libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/eh/RoadObjectMatcherObserver.kt

```diff
@@ -0,0 +1,19 @@
+package com.mapbox.navigation.core.trip.session.eh
+
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObject
+import com.mapbox.navigation.core.trip.model.roadobject.RoadObjectMatcherError
+
+/**
+ * Road objects matching listener. Callbacks are fired when matching is finished.
+ */
+interface RoadObjectMatcherObserver {
+    /**
+     * Road object matched.
+     */
+    fun onRoadObjectMatched(roadObject: RoadObject)
+
+    /**
+     * Road object matching failed.
+     */
+    fun onError(error: RoadObjectMatcherError)
```

Comment:
Can we use an `Expected` and remove `onError` callback? 🤔

Reply:
we get `Expected<RoadObject, Error>` from the native
we match the objects to SDK models
after matching  we can't build `Expected`
constructor is not public

Reply:
I think @Guardiola31337 has the `com.mapbox.navigation.ui.base.model.Expected` in mind. We will soon be migrating the project to `com.mapbox.bindgen.Expected` which exposes `com.mapbox.bindgen.ExpectedFactory` for creating objects, which is public.

Reply:
done, changed to Expected